File: HealthTestApp/app/src/main/java/com/sa/healthtest/services/ConnectCallback.java
package com.sa.healthtest.services;

import com.sa.healthtest.data.model.FitResponse;

public interface ConnectCallback {
    void successConnected(FitConnection service);
    void disconnected(FitConnection service);
    void updateFitData(FitResponse steps);
    void error(String message);
    void onPermissionDenied(FitConnection service);
}

File: HealthTestApp/app/src/main/java/com/sa/healthtest/services/ConnectCallback.kt
package com.sa.healthtest.services

import com.sa.healthtest.data.model.FitResponse

interface ConnectCallback {
        fun successConnected(service: FitConnection)
        fun disconnected(service: FitConnection)
        fun updateFitData(steps: FitResponse)
        fun error(message: String?)
        fun onPermissionDenied(service: FitConnection?)
}